ui/component/or-attribute-input/src/or-anomaly-config-chart.ts

```diff
@@ -0,0 +1,303 @@
+import "@openremote/or-chart"
+import {ChartViewConfig, OrChart, OrChartEvent} from "@openremote/or-chart"
+import {
+    Chart,
+    TimeUnit,
+    ChartDataset,
+    ChartConfiguration,
+    ScatterDataPoint,
+    TimeScaleOptions
+} from "chart.js";
+import "@openremote/or-mwc-components/or-mwc-input";
+import { customElement, property } from "lit/decorators.js";
+import { html, PropertyValues} from "lit";
+import {Console, DefaultColor4, manager, Util} from "@openremote/core";
+import moment from "moment";
+import {throttle} from "lodash";
+import {AnnotationOptions} from "chartjs-plugin-annotation";
+import {
+    AnomalyDetectionConfigObject,
+    AnomalyDetectionConfiguration,
+    AnomalyDetectionConfigurationChange,
+    AnomalyDetectionConfigurationGlobal,
+    Asset,
+    AssetDatapointQueryUnion,
+    AssetModelUtil,
+    AssetQuery,
+    Attribute,
+    AttributeRef,
+    DatapointInterval,
+    ValueDatapoint
+} from "@openremote/model";
+import {i18next, translate} from "@openremote/or-translate";
+import "@openremote/or-components/or-collapsible-panel"
+import {GenericAxiosResponse, isAxiosError} from "@openremote/rest";
+
+
+
+@customElement("or-anomaly-config-chart")
+export class OrAnomalyConfigChart extends OrChart {
+
+    @property({type: Object})
+    public attributeRef?: AttributeRef = undefined;
+    @property({type: Object})
+    public anomalyConfig?: AnomalyDetectionConfiguration = undefined;
+    @property({type: Number})
+    public timespan?: Number = undefined;
+    @property({type: String})
+    private errorMessage = ""
+    @property({type: Boolean})
+    private canRefresh = false
+
+
+    protected willUpdate(changedProps: Map<string, any>) {
+        return super.willUpdate(changedProps);
+    }
+
+    protected async _loadData() {
```

Comment:
I would try to decrease the amount of logic in this function.
I know `or-chart` does this, and it is bad, but I would try separating some of it in other functions,
so we can improve the maintainability and reusability for features like this.